src/cpp/rtps/security/SecurityManager.h

```diff
@@ -398,7 +422,7 @@ class SecurityManager
 
             EventUniquePtr event_;
 
-            uint32_t handshake_requests_sent_;
+            int handshake_requests_sent_;
```

Comment:
Yes, I agree that strictly talking it would make more sense as `uint32_t` (in fact is what I thought when i first introduced it) but in order to avoid casting in the comparison you mention, I left it as `int`, it is clearer to me.

Anyways, I defined both `max_handshake_requests` and `initial_handshake_resend_period_ms` as integers to protect against the user entering a negative value, because if we leave them as `uints`, that case would be still correct to us. But, if we leave them as `int` we could detect in a cleaner way that is out of bounds.